src/dataframe.jl

```diff
@@ -0,0 +1,159 @@
+    return DataFrame(jdf)
+end
+
+
+function Base.filter(df::DataFrame, col::Column)
+    jdf = jcall(df.jdf, "filter", JDataset, (JColumn,), col.jcol)
+    return DataFrame(jdf)
+end
+where(df::DataFrame, col::Column) = filter(df, col)
+
+function groupby(df::DataFrame, cols::Column...)
+    jgdf = jcall(df.jdf, "groupBy", JRelationalGroupedDataset,
+            (Vector{JColumn},), [col.jcol for col in cols])
+    return GroupedData(jgdf)
+end
+
+function groupby(df::DataFrame, col::String, cols::String...)
+    jgdf = jcall(df.jdf, "groupBy", JRelationalGroupedDataset,
+            (JString, Vector{JString},), col, collect(cols))
+    return GroupedData(jgdf)
+end
+
+const groupBy = groupby
+
+for func in (:min, :max,  :count, :sum, :mean)
+    @eval function $func(df::DataFrame, cols::String...)
+        jdf = jcall(df.jdf, string($func), JDataset, (Vector{JString},), collect(cols))
+        return DataFrame(jdf)
+    end
+end
+
+minimum(df::DataFrame, cols::String...) = min(df, cols...)
+maximum(df::DataFrame, cols::String...) = max(df, cols...)
+avg(df::DataFrame, cols::String...) = mean(df, cols...)
+
+
+function Base.join(df1::DataFrame, df2::DataFrame, col::Column, typ::String="inner")
+    jdf = jcall(df1.jdf, "join", JDataset,
+            (JDataset, JColumn, JString),
+            df2.jdf, col.jcol, typ)
+    return DataFrame(jdf)
+end
+
+createOrReplaceTempView(df::DataFrame, name::AbstractString) =
+    jcall(df.jdf, "createOrReplaceTempView", Nothing, (JString,), name)
+
+
+isstreaming(df::DataFrame) = Bool(jcall(df.jdf, "isStreaming", jboolean, ()))
+isStreaming(df::DataFrame) = isstreaming(df)
+
+###############################################################################
+#                                  GroupedData                                #
+###############################################################################
+
+@chainable GroupedData
+Base.show(io::IO, gdf::GroupedData) = print(io, "GroupedData()")
+
+function agg(gdf::GroupedData, col::Column, cols::Column...)
+    jdf = jcall(gdf.jgdf, "agg", JDataset,
+            (JColumn, Vector{JColumn}), col.jcol, [col.jcol for col in cols])
+    return DataFrame(jdf)
+end
+
+function agg(gdf::GroupedData, ops::Dict{<:AbstractString, <:AbstractString})
+    jmap = convert(JMap, ops)
+    jdf = jcall(gdf.jgdf, "agg", JDataset, (JMap,), jmap)
+    return DataFrame(jdf)
+end
+
+for func in (:min, :max,  :count, :sum, :mean)
+    @eval function $func(gdf::GroupedData, cols::String...)
+        jdf = jcall(gdf.jgdf, string($func), JDataset, (Vector{JString},), collect(cols))
+        return DataFrame(jdf)
+    end
+end
+
+minimum(gdf::GroupedData, cols::String...) = min(gdf, cols...)
+maximum(gdf::GroupedData, cols::String...) = max(gdf, cols...)
+avg(gdf::GroupedData, cols::String...) = mean(gdf, cols...)
+
+
+function Base.write(df::DataFrame)
+    jwriter = jcall(df.jdf, "write", JDataFrameWriter, ())
+    return DataFrameWriter(jwriter)
```

Comment:
I don't think this is right, Base.write normally does something quite different: https://docs.julialang.org/en/v1/base/io-network/#Base.write I don't know what are the conventions in Julia, but I think it would be better not to overload the built-in function with another one with a very different signature.